src/bus/spi/bus.rs

```diff
@@ -25,22 +26,19 @@ pub enum Error<SPI, CS> {
 
 pub type BUSError<SPI, CS> = bus::Error<Error<SPI, CS>>;
 
-#[cfg_attr(feature = "async-trait", async_trait::async_trait)]
-#[cfg_attr(not(feature = "async"), deasync::deasync)]
-pub trait Transfer {
-    type Error;
-    async fn transfer(&mut self, tx: &[u8], rx: &mut [u8]) -> Result<(), Self::Error>;
-}
+// pub trait Transfer {
+//     type Error;
 
-#[cfg(not(feature = "async"))]
-impl<E, T: spi::Transfer<u8, Error = E>> Transfer for T {
-    type Error = E;
+//     async fn transfer(&mut self, tx: &[u8], rx: &mut [u8]) -> Result<(), Self::Error>;
+// }
 
-    fn transfer(&mut self, tx: &[u8], rx: &mut [u8]) -> Result<(), Self::Error> {
-        rx.copy_from_slice(tx);
-        self.transfer(rx).map(|_| ())
-    }
-}
```

Comment:
This part made things very complicated when trying to feature-gate the `embedded_hal_async` impl as it was not allowing me to do it.

Another question that I have with this impl:
Why is there a `rx.copy_from_slice(tx);`?

Reply:
Simply deactivate this block when async enabled

In trait `Transfer`, rx slice and tx slice are seperated slices, which will make DMA transfer approach more easier, while in `embedded_hal::spi::Transfer` trait, rx and tx are the same slice

Reply:
> FYI: There are a few other paces that need to do the same like in the example:
> 
> https://github.com/qiuchengxuan/async-sdmmc/blob/master/sdmmc-test/src/main.rs#L64-L67

Sorry but I don't quite understand this sentence mean...

Reply:
Every place where there's an `.await` point we need to add calls `cfg` to switch between them:

Example:

https://github.com/qiuchengxuan/async-sdmmc/blob/93e58d1b98670398eb2dcaf25deb12a622fa89ac/src/bus/spi/write.rs#L36

```rust
#[cfg(feature = "async")]
self.tx(&[0xFF; 5]).await?;
#[cfg(not(feature = "async"))]
self.tx(&[0xFF; 5])?;
```
this is what I meant.

Reply:
okay